osu.Game/OsuGameBase.cs

```diff
@@ -99,6 +100,9 @@ public class OsuGameBase : Framework.Game, ICanAcceptFiles
 
         public bool IsDeployedBuild => AssemblyVersion.Major > 0;
 
+        [Cached]
+        protected readonly DifficultyRecommender DifficultyRecommender = new DifficultyRecommender();
```

Comment:
I put the recommender in base game so that only one instance is active. That's to lower API calls, I don't know how expensive calling ``GetUserRequest`` 4 times is but I can't imagine it being cheap.

Reply:
I'm thinking about again working on this PR over the weekend. I get all the other comments but this one I still would like clarified.

> resolve at necessary call sites 

I think in this PR the test, ``SongSelect``, and ``OsuGame`` are the only call sites. But the fact that ``OsuGame.PresentBeatmap`` needs access to recommender to present recommended beatmaps means that ``DifficultyRecommender`` needs to be accessed at base level. To me it made sense to cache it at base level too and you get reduction to API calls as a bonus.

Reply:
The long and short of it is that `OsuGame` and `OsuGameBase` should be as slim as they could possibly hope to be. If this can't be placed anywhere else without significant overhead, so be it, but those classes *need* to remain manageable.

Reply:
This also helps ``DifficultyRecommender`` get online faster to start making recommendations from the start, fix for #11062